File: security/src/main/java/com/nivasafinance/security/context/UserContext.java
package com.nivasafinance.security.context;

import com.nivasafinance.security.model.UserInfo;

public class UserContext {
    private static final ThreadLocal<UserInfo> threadLocal = new ThreadLocal<>();
    private static final String SYSTEM_USERNAME = "system";

    private UserContext() {
        // Private constructor to prevent instantiation
    }

    public static void setUserInfo(UserInfo userInfo) {
        threadLocal.set(userInfo);
    }

    public static UserInfo getUserInfo() {
        return threadLocal.get();
    }

    public static String getCurrentUsername() {
        UserInfo userInfo = getUserInfo();
        return userInfo != null ? userInfo.getUsername() : SYSTEM_USERNAME;
    }

    public static void clear() {
        threadLocal.remove();
    }
}

File: security/src/main/java/com/nivasafinance/security/context/UserContext.kt
package com.nivasafinance.security.context

import com.nivasafinance.security.model.UserInfo

object UserContext {
    private val threadLocal = ThreadLocal<UserInfo>()

    fun setUserInfo(userInfo: UserInfo) {
        threadLocal.set(userInfo)
    }

    fun getUserInfo(): UserInfo? {
        return threadLocal.get()
    }

    fun clear() {
        threadLocal.remove()
    }
}
